lib/LanguageServerCompletion/Handler/CompletionHandler.php

```diff
@@ -100,29 +109,42 @@ public function completion(CompletionParams $params, CancellationToken $token):
             $items = [];
             $isIncomplete = false;
             foreach ($suggestions as $suggestion) {
+                /** @var Suggestion $suggestion */
+
+                $parentNode = $this->parser
+                        ->parseSourceFile($textDocument->text)
+                        ->getDescendantNodeAtPosition($byteOffset->toInt())
+                        ->getParent();
```

Comment:
hmm, ideally we should not be parsing anything here. can this logic be in code-transform?

Reply:
Then code-transform needs to report back the "context". E.g. `CONTEXT_USE`, `CONTEXT_OTHER` or something like that. Completion has to act depending on the context. Or do you have another idea?

Reply:
Not sure I understand the `if` condtiion: if we are in a namespace declaration block, then don't add teh text edit? But why then would the completion return a class import?

Reply:
> if we are in a namespace declaration block, then don't add teh text edit?

Yes. An import doesn't make sense here.

>  But why then would the completion return a class import?

Happens here: https://github.com/phpactor/completion/blob/master/lib/Core/Completor/NameSearcherCompletor.php#L82 . Do you think this is the place to check for the namespace declaration block?

Reply:
Rather here? https://github.com/phpactor/completion/blob/master/lib/Core/Completor/NameSearcherCompletor.php#L58

i.e. we have the/a node here.

Reply:
I removed the parsing from the `CompletionHandler` and changed the `NameSearcherCompletor` to not return imports in `use` context, see phpactor/completion#58 .